### PR TITLE
fix: Collab feedback fixes

### LINKED ISSUE #7131: [Bugs] Comments feedback
## Description
- [x] If I tag someone in my very first comment(even before the bot comments on it), then the other person doesn’t get tagged
- [x] If the name of the other person isn’t set, their email id appears and it’s really long. So the side bar looks weird because email ids are so long
- [x] ~The ‘1 reply’ button in the thread doesn’t look like a button, there’s no hover state for it.~
- [x] I’m able to draw a polygon(for multi-select) even while I’m in chat mode.
- [x] ~If I leave a comment inside a tab in the tab widget, clicking the comment doesn’t open the tab~ #4886 
- [x] ~I can’t leave comments on table filters, debugger, API pane, etc. (will this be worked on?)~
- [x] When I hover on a widget, the name card shows up

### LINKED ISSUE #7153: [Bug]If the comments are already resolved and the user hover on resolve icon the tooltip displayed now is not correct
## Description

[What happened]When the comment is resolved and the user hovers on it the tooltip should say unresolve thread instead of resolve thread

### Steps to reproduce the behaviour:

image

### Important Details

- Version: [Cloud]
- OS: [e.g.MacOS]
- Browser [ chrome]
- Environment release

### PR DESCRIPTION
**Pull request by pranavkanade**
## Description
Small bug fixes for commenting feature

Fixes #7131 
Fixes #7153 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- manual

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/collab-fixes 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.43 **(0.02)** | 36.28 **(0.03)** | 33.56 **(0.01)** | 54.99 **(0.03)**
 :green_circle: | app/client/src/comments/AppComments/AppCommentThreads.tsx | 61.36 **(1.84)** | 22.73 **(0)** | 40 **(0)** | 60.47 **(1.93)**
 :red_circle: | app/client/src/comments/CommentCard/ResolveCommentButton.tsx | 93.33 **(0)** | 71.43 **(-3.57)** | 83.33 **(0)** | 92.86 **(0)**
 :green_circle: | app/client/src/components/editorComponents/WidgetNameComponent/index.tsx | 95.59 **(0.14)** | 81.03 **(0.33)** | 100 **(0)** | 95.45 **(0.14)**
 :red_circle: | app/client/src/constants/messages.ts | 72.06 **(-0.01)** | 100 **(0)** | 15.74 **(-0.05)** | 78.31 **(0.05)**
 :green_circle: | app/client/src/pages/Editor/WidgetsMultiSelectBox.tsx | 77.97 **(0.38)** | 69.23 **(1.23)** | 68.97 **(0)** | 77.98 **(0.41)**
 :green_circle: | app/client/src/pages/common/CanvasSelectionArena.tsx | 91.49 **(0.14)** | 71.98 **(0.16)** | 89.66 **(0)** | 91.21 **(0.15)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 78.23 **(1.24)** | 54.41 **(-0.14)** | 70 **(2.14)** | 84.09 **(1.59)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.71 **(0.24)** | 39.13 **(0.87)** | 36.21 **(0)** | 55.35 **(0.27)**</details>

### CODE REVIEW

**Review by sbalaji1192 — DISMISSED** [review]
LGTM

File: app/client/src/comments/AppComments/AppCommentThreads.tsx [hunk 1]
```diff
@@ -54,6 +56,7 @@ export const useSortedCommentThreadIds = (commentThreadIds: string[]) => {
       shouldShowResolved,
       appCommentsFilter,
       currentUsername,
+      unreadCommentsCount,
```
Comment (rishabhsaxena) [inline app/client/src/comments/AppComments/AppCommentThreads.tsx:59]: Think there could be still an edge case with unread count remaining the same and last updated thread is not on top?

**Review by rishabhsaxena — APPROVED** [review]

File: app/client/src/comments/AppComments/AppCommentThreads.tsx (hunk 1 above)
Comment (pranavkanade) [reply app/client/src/comments/AppComments/AppCommentThreads.tsx:59]: any suggestions on which field to use?

Comment (pranavkanade) [reply app/client/src/comments/AppComments/AppCommentThreads.tsx:62]: done

File: app/client/src/comments/AppComments/AppCommentThreads.tsx [hunk 2]
```diff
@@ -10,6 +10,8 @@ import {
   getCommentThreadsFetched,
   getSortedAndFilteredAppCommentThreadIds,
   shouldShowResolved as shouldShowResolvedSelector,
+  getLastUpdatedCommentThreadId,
+  getUnreadCommentsCount,
```
Comment (rishabhsaxena) [inline app/client/src/comments/AppComments/AppCommentThreads.tsx:14]: QQ: do we still need to subscribe to `getUnreadCommentsCount`?

**Review by rishabhsaxena — APPROVED** [review]

File: app/client/src/comments/AppComments/AppCommentThreads.tsx (hunk 2 above)
Comment (pranavkanade) [reply app/client/src/comments/AppComments/AppCommentThreads.tsx:14]: yes, it takes care of a slight delay in broadcasted values to be updated in the store.

**Review by rishabhsaxena — APPROVED** [review]

**Pull request merged**